two_sum.py

```diff
@@ -0,0 +1,51 @@
+'''
+Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.
+
+You may assume that each input would have exactly one solution, and you may not use the same element twice.
+
+You can return the answer in any order.
+
+ 
+
+Example 1:
+
+Input: nums = [2,7,11,15], target = 9
+Output: [0,1]
+Explanation: Because nums[0] + nums[1] == 9, we return [0, 1].
+Example 2:
+
+Input: nums = [3,2,4], target = 6
+Output: [1,2]
+Example 3:
+
+Input: nums = [3,3], target = 6
+Output: [0,1]
+'''
+
+def two_sum(nums, target):
+    num_indices = {}
+    for i, num in enumerate(nums):
+        complement = target - num
+        if complement in num_indices:
+            return [num_indices[complement], i]
+        num_indices[num] = i
```

Comment:
Your current implementation for the two-sum problem looks efficient, but it doesn't handle the case where no solution is found. The function should return `None`. See here: https://chat.openai.com/share/f3967d74-3668-4eb4-a547-7bf4ee05d617